### DISCUSSION TITLE
API for other extensions?

### DISCUSSION THREAD

**Discussion by Jak2k** [uBlockOrigin/uBlock-issues#3244] [Q&A] [answered]
I'm currently making a browser extension that calculates a score for the currently viewed website. I would like to access the number of blocked things in that tab to use it in that calculation. Is there any way to get this data?

**Comment by iam-py-test** [comment]
No, it is not supported by uBo, and I believe gorhill has declined such a feature in the past. It would be difficult to implement.
The number of blocked requests is not an accurate metric for assessing the privacy of a website.
Firstly, not every blocked request is a tracker: blocked requests may be cookie consent forms, ads, or even false positives.
Secondly, each blocked request is not an individual tracker. One tracker may have several components, or may keep trying to reach out to its server, causing a large number of blocked requests.
Moreover, your extension would be reliant on uBo bring installed.
Lastly, this would cease to be possible in MV3, which is coming to Chromium browsers in a matter of weeks (I think?).
Thank you

**Reply by Jak2k** [reply]
I'm interested in all things that get blocked. The extension wouldn't be reliant as I could just add a fallback.

MV3 isn't a problem for me,because I only develop for Firefox.

**Comment by gorhill** [comment, marked as answer]
You can use [`webRequest.onErrorOccurred`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onErrorOccurred) to find out what was blocked. It's what [uBO Scope uses](https://github.com/gorhill/uBO-Scope/blob/0.1.12/js/start.js#L1129), no need to hook into other extensions.
